### DISCUSSION TITLE
telemetry issue with TX16S radio and Express LRS 2.4  only 10 sensors

### DISCUSSION THREAD

**Discussion by canalrodrigo** [ExpressLRS/ExpressLRS#1423] [Q&A] [answered]
With TX16S radio and Express LRS 2.4 , I can only get it to find 10 sensors: (1RSS, 2RSS, RQly, RSNR, ANT, RFMD, TPWR, TRSS, TQly, TSNR) I already tried updating the RX and TX to the latest firmware, with the telemetry option, I also deleted all sensor and discover new sensor, and also increase the Packet Rates but still don’t get the rest of the sensors

**Comment by SunjunKim** [comment]
Did you enable the TELEMETRY option in BetaFlight?
<img width="601" alt="Screen Shot 2022-02-11 at 2 06 09 PM" src="https://user-images.githubusercontent.com/12195507/153540346-9a988d29-35ef-44c3-822f-817ed8b4d7c6.png">

**Reply by canalrodrigo** [reply]
Yes I did

**Reply by SunjunKim** [reply]
* Are your RX/TX wires attached to your FC correctly? Do you get control signals?
* Have you waited enough while `discover new sensors`?

**Reply by canalrodrigo** [reply]
1.- I think the wires are attached correctly, since I bought the bind and fly Diatone GT R349 HD, and I checked with the FC manual/diagram 
2.-Yes. I waited enough 
![MAMBA F405 Mini MK2](https://user-images.githubusercontent.com/78128464/153542678-307e35b3-3f92-45fb-be34-937a1fa21e53.jpg)

**Reply by canalrodrigo** [reply]
[BTFL_cli_R349HD_20220210_233227.txt](https://github.com/ExpressLRS/ExpressLRS/files/8046153/BTFL_cli_R349HD_20220210_233227.txt)

**Reply by SunjunKim** [reply]
I found no RX_SERIAL in your BF configuration.
Check https://oscarliang.com/setup-expresslrs-2-4ghz/#betaflight-setup-for-expresslrs-receiver

**Reply by canalrodrigo** [reply]
![image](https://user-images.githubusercontent.com/78128464/153543574-c1ba1fed-524e-4cf1-bba3-d2368392ab5e.png)

**Reply by SunjunKim** [reply]
I can't find in your `diff all`log:
```
# serial
serial 2 8192 115200 57600 0 115200
serial 5 2 115200 57600 0 115200
```

+ Following the diagram you uploaded (+assume you followed Crossfire Nano diagram), it should be UART6.

**Reply by SunjunKim** [reply]
+ UART1 is only compatible with inverted UART signal and no RX pin available.
~~which cannot be used with ELRS. You must not have RC signals as well (check in the receiver tap)~~

Please attach your **exact** wiring configuration to spot out the problem.

**Reply by SunjunKim** [reply]
Do you have GPS as well? It's occupying UART6 currently.

**Reply by canalrodrigo** [reply]
That’s strange, because that’s how I fly daily this quad, I do have RC signals in betaflight
![20220210_235855](https://user-images.githubusercontent.com/78128464/153549332-03bb92cf-f5ed-4462-bf87-490bc838916c.jpg)

**Reply by canalrodrigo** [reply]
Do you have GPS as well? It's occupying UART6 currently.
R: that's correct

**Reply by SunjunKim** [reply]
Ah OK.
Did you compile RX firmware with the `RCVR_INVERT_TX` option, and I guess you've connected the receiver with three wires (gnd, 5v, and sbus)?

There's no TX1 pinout available on your board. You cannot use telemetry with this configuration.

**Reply by SunjunKim** [reply]
Alternatively, you can move your RX to UART 3 (TX3/RX3), and move your VTX control to a pin something like buzzer or LED, and set a softserial there.

**Comment by SunjunKim** [comment, marked as answer]
To make all the things clear and prevent guesswork of troubleshooting, I'd suggest you take pictures of your wiring around the FC, and indicate which wire goes where clearly.

Your problem is not having a full-duplex UART1 from FC to RX. You need four wires fully connected between TX and RX to get the telemetry information.

**Reply by canalrodrigo** [reply]
perfect, thanks for your help, I'll do that and atatach the pictures

**Reply by GrinFactor** [reply]
Please could someone explain how to solve the issue. I have a T16 and using a Happymodel Exprsslrs EP1 receiver. Only seeing 10x sensors. 
I'm using the Zeus F722 board that states I must connect to UART 1 (which I have). How can I get full sensor readout to my T16?

**Reply by jklong03** [reply]
On my Zeus F722 Mini (with ELRS 3.0RC1, BF 4.3.0  and EdgeTX 2.6.0 on my TX-16S)  I am seeing 17 sensors. Bear in mind that even at the faster packet rates, with the longer telemetry ratios it can take longer than you might think to see some values update.  I believe that some values are queued up to be transmitted back down only when their values change. Is your counting of only 10 based on a "discover new sensors" after you have deleted all sensors first?

**Comment by canalrodrigo** [comment]
here are some pictures I'll make some tags for the conection
![20220211_004332](https://user-images.githubusercontent.com/78128464/153549081-7e87a0cc-0790-42a4-8e42-93936c3a0d21.jpg)
![20220211_004550](https://user-images.githubusercontent.com/78128464/153549085-b6727582-0cc6-4e16-91ca-6b45c1456851.jpg)
![20220211_004613](https://user-images.githubusercontent.com/78128464/153549087-374e8010-efb2-4ef9-8a2c-6fd4f02e1a6f.jpg)
![20220211_004643](https://user-images.githubusercontent.com/78128464/153549092-53afd4bc-4078-4c1c-a87b-5f350e17a44e.jpg)
![20220211_004709](https://user-images.githubusercontent.com/78128464/153549098-b0df5a11-89f2-43b1-9d18-71926c820e90.jpg)
![20220211_002918](https://user-images.githubusercontent.com/78128464/153549101-c5b4c469-f9cd-48fc-9d3d-74227e930519.jpg)
s

**Reply by SunjunKim** [reply]
Even though the white wire is labeled as "TX1", the wiring diagram you posted above mentions this as "PPM" which doesn't seem to act like that.

Conventionally, PPM and SBUS are all connected to RX1, except SBUS is attached with an inverter.

**Reply by SunjunKim** [reply]
+ Your antenna wires are bent considerably and placed nearby metal objects. 
It will be OK, but you should not expect the full range from that wire management.

**Reply by canalrodrigo** [reply]
Even though the white wire is labeled as "TX1", the wiring diagram you posted above mentions this as "PPM" which doesn't seem to act like that.
your appreciation is correct, having that in mind what would you recommend me to do? 

Your antenna wires are bent considerably and placed nearby metal objects
I'm not to worry about that since I lost vtx signal way before tx/rx (my VTX is only 400mw)
[resource BTFL_cli_R349HD_20220211_124334.txt](https://github.com/ExpressLRS/ExpressLRS/files/8050950/resource.BTFL_cli_R349HD_20220211_124334.txt)

**Comment by IvanBiv** [comment]
I had a similar problem: I only saw 10 sensors in telemetry. I cleared the Softserial2 assignment and found 23 sensors.
FC: Matek F411 WSE
![image](https://user-images.githubusercontent.com/7404103/185079560-4ffcb719-2377-4153-af3a-17a65ef51e99.png)

**Comment by canalrodrigo** [comment]
Thanks for the advice!!

**Comment by mixa1977** [comment]
Hello. Same problem. Tx16s+crsf matek 405se. Did you manage to solve?

**Reply by mixa1977** [reply]
I solved the problem. Softserial1 set "disabled"

**Reply by IvanBiv** [reply]
Excellent!  To solve the problem with the lack of telemetry readings on the equipment, you need to check the documentation of your flight controller in terms of softserial.  Probably something from softserial needs to be disabled.
